Specs/Scene/I3dmParserSpec.js

```diff
@@ -0,0 +1,41 @@
+import { I3dmParser } from "../../Source/Cesium.js";
+import Cesium3DTilesTester from "../Cesium3DTilesTester.js";
+import createScene from "../createScene.js";
+
+describe("Scene/I3dmParser", function () {
+  var scene;
+
+  beforeAll(function () {
+    scene = createScene();
+
+    // Keep the error from logging to the console when running tests
+    spyOn(I3dmParser, "_deprecationWarning");
+  });
+
+  afterAll(function () {
+    scene.destroyForSpecs();
+  });
+
+  afterEach(function () {
+    scene.primitives.removeAll();
+  });
+
+  it("throws with undefined arrayBuffer", function () {
+    expect(function () {
+      I3dmParser.parse(undefined);
+    }).toThrowDeveloperError();
+  });
+
+  it("throws with empty gltf", function () {
+    // Expect to throw error due to invalid glTF magic.
+    var arrayBuffer = Cesium3DTilesTester.generateInstancedTileBuffer();
+    Cesium3DTilesTester.loadTileExpectError(scene, arrayBuffer, "i3dm");
+  });
+
+  it("throws with unsupported I3DM version", function () {
+    var arrayBuffer = Cesium3DTilesTester.generateInstancedTileBuffer({
+      version: 2,
+    });
+    Cesium3DTilesTester.loadTileExpectError(scene, arrayBuffer, "i3dm");
+  });
```

Comment:
Could we add at least one test of parsing a valid I3DM?

Reply:
I've added a few rendering and picking tests, I hope that, along with the list of loading test cases, covers all the bases here.

Reply:
@sanjeetsuhag The tests in this file could definitely use some work. `I3dmParserSpec` should be testing `I3dmParser` directly, not indirectly via `Instanced3DModel3DTileContent`. If you were to run `I3dmParserSpec` by itself it would result in pretty poor coverage over `I3dmParser`.